src/js/Attachment/__test__/onDragEnter.test.ts

```diff
@@ -0,0 +1,110 @@
+import Attachment from '../index';
+import {createEvent, fireEvent} from '@testing-library/dom';
+
+describe('Unit test Attachment onDragEnter', () => {
+  beforeEach(() => {
+    jest.spyOn(console, 'error');
+    // @ts-ignore
+    console.error.mockImplementation(() => { });
+  });
+  afterEach(() => {
+    // @ts-ignore
+    console.error.mockRestore();
+  });
+
+  test('onDragEnter is fired successfully with dataTransfer.items[].kind', () => {
+    const myAttachment = new Attachment();
+    const container = myAttachment.render();
+    const drag = container.getElementsByClassName('kuc-attachment-file');
+
+    const newFile = new File([''], 'file.png', {
+      type: 'image/png',
+    });
+    Object.defineProperty(newFile, 'kind', {
+      value: 'file'
+    });
+
+    const dragEvent = createEvent.dragEnter(drag[0]);
+    Object.defineProperty(dragEvent, 'dataTransfer', {
+      value: {
+        items: [newFile]
+      },
+    });
+    try {
+      fireEvent(drag[0], dragEvent);
+    } catch (error) {
+      expect(error).toBeFalsy();
+    }
+  });
+
+  test('onDragEnter is fired successfully with dataTransfer.types', () => {
+    const myAttachment = new Attachment();
+    const container = myAttachment.render();
+    const drag = container.getElementsByClassName('kuc-attachment-file');
+
+    const newFile = new File([''], 'file.png', {
+      type: 'image/png',
+    });
+    Object.defineProperty(newFile, 'kind', {
+      value: '123'
+    });
+
+    const dragEvent = createEvent.dragEnter(drag[0]);
+    Object.defineProperty(dragEvent, 'dataTransfer', {
+      value: {
+        items: [newFile],
+        types: ['files']
+      },
+    });
+    try {
+      fireEvent(drag[0], dragEvent);
+    } catch (error) {
+      expect(error).toBeFalsy();
+    }
+  });
+
+  test('onDragEnter will not fire without dataTransfer.items and dataTransfer.types', () => {
+    const myAttachment = new Attachment();
+    const container = myAttachment.render();
+    const drag = container.getElementsByClassName('kuc-attachment-file');
+
+    const dragEvent = createEvent.dragEnter(drag[0]);
+    Object.defineProperty(dragEvent, 'dataTransfer', {
+      value: {}
+    });
+    try {
+      fireEvent(drag[0], dragEvent);
+    } catch (error) {
+      expect(error).toBeFalsy();
+    }
+  });
+
+  test('onDragEnter will not fire without valid types', () => {
+    const myAttachment = new Attachment();
+    const container = myAttachment.render();
+    const drag = container.getElementsByClassName('kuc-attachment-file');
+
+    const newFile = new File([''], 'file.png', {
+      type: 'image/png',
+    });
+    Object.defineProperty(newFile, 'kind', {
+      value: '123'
+    });
+
+    const dragEvent = createEvent.dragEnter(drag[0]);
+    Object.defineProperty(dragEvent, 'dataTransfer', {
+      value: {
+        items: [newFile],
+        types: ['123']
+      },
+    });
+    try {
+      fireEvent(drag[0], dragEvent);
```

Comment:
Seem miss some expectation.

Reply:
This is to verify that `Attachment` component won't crashed (internal function handle that)